### PR TITLE
feat: add clear selection button to multiselect

### PR DESCRIPTION
**Pull request by massao**
# Purpose of PR

Adds `onClearSelection` prop to multiselect that makes a clear selection button show if there are more than 1 element selected.

Added a new Story to storybook that uses the `onClearSelection`

image

image

### CODE REVIEW

**Review by Lelith — APPROVED** [review]
i think code wise it is good, just want to sync with @damann about the hover styles

**Review by cf-remylenoir — APPROVED** [review]
It looks good 👍

File: packages/components/multiselect/src/Multiselect.test.tsx [hunk 1]
```diff
@@ -277,3 +277,49 @@ describe('Multiselect with select all', () => {
...
+      true,
+    );
+
+    expect(
+      within(screen.getByTestId('cf-multiselect-current-selection')).getByText(
+        'Avocado 🥑',
+      ),
+    ).toBeInTheDocument();
+    expect(screen.queryByLabelText('Clear selection')).not.toBeInTheDocument();
+  });
```
Comment (miguelcrespo) [inline packages/components/multiselect/src/Multiselect.test.tsx:302]: suggestion: can you add a test for when there's nothing selected the button does not get rendered?

File: packages/components/multiselect/src/Multiselect.tsx [hunk 2]
```diff
@@ -172,6 +179,11 @@ function _Multiselect(props: MultiselectProps, ref: React.Ref<HTMLDivElement>) {
 
   const internalListRef = useRef<HTMLUListElement>(null);
 
+  const showClearButton = useMemo(
+    () => currentSelection.length > 1 && typeof onClearSelection === 'function',
+    [currentSelection.length, onClearSelection],
+  );
+
```
Comment (miguelcrespo) [inline packages/components/multiselect/src/Multiselect.tsx:186]: suggestion: this doesn't need to be memoized, it's a constant operation

**Review by miguelcrespo — APPROVED** [review]

**Pull request merged**